### PR TITLE
[NTH-64] create Model Appointments 

### PR DESCRIPTION
**Pull request by raghdarizq**
- The code represents a React component for an appointments modal form.
- It uses Material-UI components for styling and form inputs.
- The useState hook is used to manage the state of time inputs.
- The formik hook handles form management, including validation and submission.
- The modal displays the form and can be opened or closed based on the "open" prop.
- The form includes date and time inputs using DatePicker and TimePicker components.
- Validation messages are displayed for the inputs based on formik errors and touched fields.
- Time inputs can be dynamically added or deleted.
- Buttons are provided for canceling and confirming the form submission.
- The component provides a user-friendly interface for selecting and scheduling availability time slots.

### CODE REVIEW

File: src/components/AppointmentsModal/classes.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+interface Type {
+  [key: string]: string;
+}
+
+const Icon: Type = {
+  fontSize: '20px',
+  cursor: 'pointer',
+  color: '#888888',
+  // marginLeft: ' 18px',
```
Comment (ahmedmurtaja) [inline src/components/AppointmentsModal/classes.ts]: remove commented code

File: src/components/AppointmentsModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      aria-labelledby="transition-modal-title"
+      aria-describedby="transition-modal-description"
+      open={open}
+      onClose={handleClose}
+      closeAfterTransition
+
+    >
+      <Fade in={open}>
+        <Box sx={style} component="form" onSubmit={formik.handleSubmit}>
+          <Typography component="b">Choose Your Availability Times </Typography>
```
Comment (ahmedmurtaja) [inline src/components/AppointmentsModal/index.tsx:66]: i think using heading elements ``h1,h2,...``is better than using `` b``

File: src/pages/therapistProfile/schema.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          .min(yup.ref('fromDate'), 'To date should be after from date')
+          .required('From date is required'),
+        to: yup.date()
+          .min(yup.ref('fromTime'), 'To time should be after from time')
+          .required('To date is required'),
+      }),
+    )
+    .required('Time array is required'),
+});
+  // to: yup
```
Comment (ahmedmurtaja) [inline src/pages/therapistProfile/schema.ts]: commented code

**Review by ahmedmurtaja — CHANGES_REQUESTED** [review]

File: src/components/AppointmentsModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+                                      FormikErrors<{ from: string; to: string; }>)?.from),
+                                  onBlur: formik.handleBlur,
+
+                                },
+                              }}
+                            />
+                          </LocalizationProvider>
+                        </Grid>
+                        <Grid item xs={timeInput.length === 1 ? 6 : 5}>
+                          <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (aaamra) [inline src/components/AppointmentsModal/index.tsx:150]: duplicated

File: src/components/AppointmentsModal/index.tsx (hunk 2 above)
Comment (raghdarizq) [reply src/components/AppointmentsModal/index.tsx:66]: It is not appropriate

**Review by ahmedmurtaja — APPROVED** [review]

**Review by YasserAmro0 — APPROVED** [review]

**Review by MohannadSabaa — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**